templates/pkg/resource/manager.go.tpl

```diff
@@ -252,6 +258,53 @@ func (rm *resourceManager) IsSynced(ctx context.Context, res acktypes.AWSResourc
 	return true, nil
 }
 
+// EnsureTags ensures that tags are present inside the AWSResource.
+// If the AWSResource does not have any existing resource tags, the 'tags'
+// field is initialized and the controller tags are added.
+// If the AWSResource has existing resource tags, then controller tags are
+// added to the existing resource tags without overriding them.
+// If the AWSResource does not support tags, only then the controller tags
+// will not be added to the AWSResource.
+func (rm *resourceManager) EnsureTags(
+    ctx context.Context,
+    res acktypes.AWSResource,
+    md acktypes.ServiceControllerMetadata,
+) error {
+{{- if $hookCode := Hook .CRD "ensure_tags" }}
+{{ $hookCode }}
+{{ else }}
+{{ $tagField := .CRD.GetTagField -}}
+{{ if $tagField -}}
```

Comment:
`GetTagField` should panic if one can't be found, right? We don't want to generate an empty method here, otherwise? Therefore, we should be able to remove this condition

Reply:
We wanna return a method with just `return nil` for the resources that do not support tagging. For example: apigatewayv2's route resource.

GetTagField will return `nil` when generator.yaml contains `tags: ignore: true` and then `EnsureTags` will only contain `return nil` statement.